elements/lisk-p2p/src/peer_book/base_list.ts

```diff
@@ -13,146 +13,195 @@
  *
  */
 import { ExistingPeerError } from '../errors';
-import { P2PPeerInfo } from '../p2p_types';
-import { evictPeerRandomlyFromBucket, getBucketId, PEER_TYPE } from '../utils';
+import { P2PEnhancedPeerInfo, P2PPeerInfo } from '../p2p_types';
+import {
+	evictPeerRandomlyFromBucket,
+	getBucketId,
+	PEER_TYPE,
+	sanitizeEnhancedPeerInfo,
+} from '../utils';
 
 export interface PeerListConfig {
-	readonly peerBucketCount: number;
-	readonly peerBucketSize: number;
+	readonly numOfBuckets: number;
+	readonly bucketSize: number;
 	readonly secret: number;
 	readonly peerType: PEER_TYPE;
 }
 
-export interface CustomPeerInfo {
-	readonly peerInfo: P2PPeerInfo;
-	readonly dateAdded: Date;
+export type Bucket = Map<string, P2PEnhancedPeerInfo>;
+
+export interface BucketInfo {
+	readonly bucketId: number;
+	readonly bucket: Bucket;
 }
 
-// Base list class is covering a basic peer list that has all the functionality to handle buckets with default eviction strategy
 export class BaseList {
-	protected peerMap: Map<number, Map<string, CustomPeerInfo>>;
+	protected bucketIdToBucket: Map<number, Bucket>;
```

Comment:
Prefer `bucketIdToPeersMap`? Looks a bit weird with bucket word used 2 times

Reply:
It is named exactly what it is: a map of `bucketId` to `Bucket` type.